kratos/future/solving_strategies/builders/builder.h

```diff
@@ -0,0 +1,420 @@
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "utilities/amgcl_csr_conversion_utilities.h"
+#include "utilities/amgcl_csr_spmm_utilities.h"
+#include "utilities/builtin_timer.h"
+#include "utilities/dof_utilities/dof_array_utilities.h"
+#include "utilities/timer.h"
+
+namespace Kratos::Future
+{
+
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+//FIXME: We should place this somewhere else
+/**
+ * @brief Auxiliary container to store the linear system
+ * This auxiliary container is intended to store all the arrays requires for the linear system setup
+ * @tparam TSparseMatrixType The sparse matrix type
+ * @tparam TSystemVectorType The system vector type
+ */
+template <class TSparseMatrixType, class TSystemVectorType>
+struct LinearSystemContainer
+{
+    typename TSparseMatrixType::Pointer pLhs = nullptr; // Pointer to the LHS matrix
+
+    typename TSystemVectorType::Pointer pRhs = nullptr; // Pointer to the RHS vector
+
+    typename TSystemVectorType::Pointer pDx = nullptr; // Pointer to the solution increment vector
+
+    typename TSparseMatrixType::Pointer pEffectiveLhs = nullptr; // Pointer to the effective LHS matrix (i.e., after applying system constraints)
+
+    typename TSystemVectorType::Pointer pEffectiveRhs = nullptr; // Pointer to the effective RHS vector (i.e., after applying system constraints)
+
+    typename TSystemVectorType::Pointer pEffectiveDx = nullptr; // Pointer to the effective solution increment vector (i.e., after applying system constraints)
+
+    typename TSparseMatrixType::Pointer pEffectiveT = nullptr; // Linear system constraints total relation matrix
+
+    typename TSparseMatrixType::Pointer pConstraintsT = nullptr; // Master-slave constraints relation matrix
+
+    typename TSystemVectorType::Pointer pConstraintsQ = nullptr; // Master-slave constraints constant vector
+
+    typename TSparseMatrixType::Pointer pDirichletT = nullptr; // Dirichlet constraints relation matrix
+
+    typename TSparseMatrixType::Pointer pMassMatrix = nullptr; // Pointer to the mass matrix
+
+    typename TSparseMatrixType::Pointer pDampingMatrix = nullptr; // Pointer to the damping matrix
+
+    void Clear()
+    {
+        if (pLhs != nullptr) {
+            pLhs->Clear();
+        }
+        if (pRhs != nullptr) {
+            pRhs->Clear();
+        }
+        if (pDx != nullptr) {
+            pDx->Clear();
+        }
+        if (pEffectiveLhs != nullptr) {
+            pEffectiveLhs->Clear();
+        }
+        if (pEffectiveRhs != nullptr) {
+            pEffectiveRhs->Clear();
+        }
+        if (pEffectiveDx != nullptr) {
+            pEffectiveDx->Clear();
+        }
+        if (pEffectiveT != nullptr) {
+            pEffectiveT->Clear();
+        }
+        if (pConstraintsT != nullptr) {
+            pConstraintsT->Clear();
+        }
+        if (pConstraintsQ != nullptr) {
+            pConstraintsQ->Clear();
+        }
+        if (pDirichletT != nullptr) {
```

Comment:
@KratosMultiphysics/technical-committee:
1. Check if we actually need this to be here or we can store it in the elimination builder
2. If we keep it in here, put a comment

Reply:
As we discussed, we can store it in the elimination builder, which is the place in which we actually need it.